### PR TITLE
Updated HDF5 tools page to provide more detailed listing of tools

### LINKED ISSUE #81: Update HDF5 tools page
Update https://nwb-overview.readthedocs.io/en/latest/tools/hdfview/hdfview.html to provide a list general HDF5 tools that can be useful or inspecting NWB HDF5 file. E.g,:

* HDFView
* HDF5 Command Line Tools, e.g., h5dump, h5ls etc. see https://portal.hdfgroup.org/display/HDF5/HDF5+Command-line+Tools
    * At least some of those tools also support S3 `` h5ls --vfd=ros3 -r <s3path>`` or on Windows ``h5ls --vfd=ros3 --s3-cred="(,,)" <s3path>``
* In VS Code, you can browse NWB files in VS Code using this extension https://marketplace.visualstudio.com/items?itemName=h5web.vscode-h5web and adding an editor association for *.nwb
* In Jupyter the following may be useful
      * https://github.com/jupyterlab/jupyterlab-hdf5 
      * https://github.com/European-XFEL/h5glance
      * https://github.com/silx-kit/jupyterlab-h5web
*

### PR DESCRIPTION
**Pull request by oruebel**
Fix  #81

This PR makes the page describing general HDF5 tools more general to also include information about plugins for Jupyter, VSCode and in general provide more details about available tools. Since the page is no longer specific to HDFView I also renamed to page to avoid possible confusion.

### CODE REVIEW

**Comment by oruebel** [comment]
Here the link to the main page that this PR updates: 

https://nwb-overview--85.org.readthedocs.build/en/85/tools/hdftools/hdftools.html#analysistools-hdftools

File: docs/source/tools/hdftools/hdftools.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
+.. _analysistools-hdftools:
+
+HDF Tools
+---------
+
+.. short_description_start
+
+:ref:`analysistools-hdftools` as part of the HDF5 suite and the broader HDF community
+provide a broad range of useful tools for inspecting and browsing HDF5 files. For example
```
Comment (bendichter) [inline docs/source/tools/hdftools/hdftools.rst:9]:
```suggestion
There are a broad range of useful tools for inspecting and browsing HDF5 files. For example,
```

Comment (oruebel) [reply docs/source/tools/hdftools/hdftools.rst:9]: Thanks for the suggestion. We need the ``:ref:`analysistools-hdftools`` in the text as this how we navigate from the page with listing of all the tools to the individual pages. I'll see what I can do to smooth out that sentence without loosing the link. I think maybe we can just use your sentence and "HDF5 Tools: " as a title prefix.

File: docs/source/tools/hdftools/hdftools.rst [hunk 2]
```diff
@@ -0,0 +1,148 @@
+.. _analysistools-hdftools:
+
+HDF Tools
+---------
+
+.. short_description_start
+
+There are a broad range of useful tools for inspecting and browsing HDF5 files. For example,
```
Comment (oruebel) [inline docs/source/tools/hdftools/hdftools.rst:8]:
```suggestion
:ref:`analysistools-hdftools`: There are a broad range of useful tools for inspecting and browsing HDF5 files. For example,
```

**Review by bendichter — APPROVED** [review]

**Review by rly — APPROVED** [review]

**Comment by rly** [comment]
I'll add an image for the VS Code extension

**Review by rly — APPROVED** [review]

**Pull request merged**